### PR TITLE
a very simple sqlite datastore for Pod/Session/Container

### PR DESCRIPTION
**Pull request by guowenjian90**
a very simple simple sqlite datastore for Pod/Session/Container
currently, it convert these objects into json object and save it as a text, in future, we can use better faster encoding method like protobuf.

### CODE REVIEW

File: pkg/nodeagent/store/factory/storefactory.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	if v, ok := obj.(store.Container); !ok {
+		return nil, fmt.Errorf("%v not a store.Container object", obj)
+	} else {
+		return &v, nil
+	}
+}
+
+func (s *ContainerStore) PutContainer(container *store.Container) error {
+	err := s.PutObject(container.Identifier, container)
+	if err != nil {
```
Comment (h-w-chen) [inline pkg/nodeagent/store/factory/storefactory.go:122]: nit: return s.PutObject oneliner

File: pkg/nodeagent/test/storefactory.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"time"
+
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store"
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store/factory"
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store/sqlite"
+	_ "github.com/mattn/go-sqlite3"
+	v1 "k8s.io/api/core/v1"
+)
+
+func main() {
```
Comment (h-w-chen) [inline pkg/nodeagent/test/storefactory.go:14]: how is this test (main) invoked? is this a manual test case?

File: pkg/nodeagent/store/factory/storefactory.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+type SessionStore struct {
+	store.Store
+}
+
+type ContainerStore struct {
+	store.Store
+}
+
+func NewPodSqliteStore(options *sqlite.SQLiteStoreOptions) (*PodStore, error) {
```
Comment (h-w-chen) [inline pkg/nodeagent/store/factory/storefactory.go:38]: do we have to embed sqlite in the func name?

File: pkg/nodeagent/store/factory/storefactory.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+		func(obj interface{}) (string, error) { return store.JsonFromPod(obj.(*store.Pod)) }); err != nil {
+		return nil, err
+	} else {
+		return &PodStore{store}, nil
+	}
+}
+
+func (s *PodStore) GetPod(identifer string) (*store.Pod, error) {
+	obj, err := s.GetObject(identifer)
+	if err != nil {
```
Comment (h-w-chen) [inline pkg/nodeagent/store/factory/storefactory.go:50]: type conversion code seems repetitive in other places too. Is it possible to leverage golang generics?

**Review by h-w-chen — CHANGES_REQUESTED** [review]
left a few comments; waiting for author  to address them

File: pkg/nodeagent/store/factory/storefactory.go (hunk 3 above)
Comment (guowenjian90) [reply pkg/nodeagent/store/factory/storefactory.go:38]: this is meant to be Sqlite store, we want to use other underlying store tech in future.

File: pkg/nodeagent/test/storefactory.go (hunk 2 above)
Comment (guowenjian90) [reply pkg/nodeagent/test/storefactory.go:14]: yes, I have not make unit test, thought this store is simply put/get object, better to test it with real data.

File: pkg/nodeagent/store/factory/storefactory.go (hunk 4 above)
Comment (h-w-chen) [reply pkg/nodeagent/store/factory/storefactory.go:50]: golang 1.18+ has generic; not mandatory to use it here; just in case it is useful.

**Review by h-w-chen — APPROVED** [review]
lgtm in general; a few commends not yet addressed are not blocking

**Pull request merged**